api/python/quilt3/data_transfer.py

```diff
@@ -447,14 +447,8 @@ def _calculate_etag(file_path):
 def delete_object(bucket, key):
     s3_client = create_s3_client()
 
-    if key.endswith('/'):
-        paginator = s3_client.get_paginator('list_objects_v2')
-        for response in paginator.paginate(Bucket=bucket, Prefix=key):
-            for obj in response.get('Contents', []):
-                s3_client.delete_object(Bucket=bucket, Key=obj['Key'])
-    else:
-        s3_client.head_object(Bucket=bucket, Key=key)  # Make sure it exists
-        s3_client.delete_object(Bucket=bucket, Key=key)  # Actually delete it
+    s3_client.head_object(Bucket=bucket, Key=key)  # Make sure it exists
```

Comment:
Why do we need to head instead of just handling the exception if the delete fails?

Reply:
Cause delete doesn't fail if the object doesn't exist... it's just a no-op.